torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -404,6 +404,20 @@ void ProcessGroupNCCL::parseNcclBlockingWait() {
   }
 }
 
+void ProcessGroupNCCL::parseNcclAsyncErrorHandling() {
```

Comment:
This function looks good. Wondering if we can reuse parseNcclBlockingWait() somehow

Reply:
I think it should be doable, I've made an issue to follow up on this: #44205 (mainly because we might be able to make this helper function more broadly available in the PyTorch codebase)